### PR TITLE
Added `validateDependenciesVersions()` function as a part of the new release process.

### LINKED ISSUE #13973: Release process: Checks before publishing (validate the release)
## 📝 Provide a description of the new feature

**🎯 The goal is to verify if all CKEditor 5 packages are specified and require themselves to use the same version.**

- Example:
    - Release: 38.0.0
        - `engine#version`: 38.0.0, `engine#depenedencies#ckeditor5-core`: ^37.0.0 - NOK
        - `engine#version`: 38.0.0, `engine#depenedencies#ckeditor5-core`: ^38.0.0 - OK
    - Release: 0.0.0-nightly-20230425.0
        - `engine#version`: 0.0.0-nightly-20230425.0, `engine#dependencies#ckeditor5-core`: ^0.0.0-nightly-20230425.0 - OK
        - `engine#version`: 0.0.0-nightly-20230425.0, `engine#dependencies#ckeditor5-core`: `@nightly` - NOK
        - `engine#version`: 0.0.0-nightly-20230425.0, `engine#dependencies#ckeditor5-core`: ^37.0.0 - NOK
- Possible arguments:
	- a version (string)
	- a directory where to look for packages (string)
---

If you'd like to see this feature implemented, add a 👍 reaction to this post.

### PR DESCRIPTION
**Pull request by martnpaneq**
### Suggested merge commit message ([convention](https://ckeditor.com/docs/ckeditor5/latest/framework/contributing/git-commit-message-convention.html))

Internal: Added `validateDependenciesVersions()` function as a part of the new release process. Closes #13973.

---

### Additional information

_For example – encountered issues, assumptions you had to make, other affected tickets, etc._

### CODE REVIEW

**Review by psmyrek — CHANGES_REQUESTED** [review]
It looks like it does not work correctly, because it compares versions strictly:

obraz

**Comment by martnpaneq** [comment]
We agreed that the dependency version should not be an exact match with the argument `version`, but rather we should check if it matches semantically following semver rules.
For example if the argument is `37.0.0`, and the dependency version is `^37.0.0` or `~37.0.0`, no errors should be reported.

File: scripts/release/validatedependenciesversions.js [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+const chalk = require( 'chalk' );
+const semver = require( 'semver' );
+const { normalizeTrim } = require( 'upath' );
+const isCKEditor5Package = require( './isckeditor5package' );
+
+/**
+ * Validates if the versions of packages and their dependencies in specified directory match the provided version.
+ *
+ * @param {Object} options
+ * @param {String} releaseDirectory Path to directory with packages to validate.
```
Comment (psmyrek) [inline scripts/release/validatedependenciesversions.js:23]:
```suggestion
 * @param {String} options.releaseDirectory Path to directory with packages to validate.
```

File: scripts/release/validatedependenciesversions.js [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+ * @param {String} options.version Version that all packages and their dependencies need to match.
+ */
+module.exports = async function validateDependenciesVersions( { releaseDirectory, version } ) {
```
Comment (psmyrek) [inline scripts/release/validatedependenciesversions.js:26]: In other scripts we use the `packagesDirectory` name for the place where all packages to process are located, or `packagesDirectory` + `outputDirectory` if both locations are needed. What do you think about changing the name `releaseDirectory` → `packagesDirectory`?

Comment (martnpaneq) [reply scripts/release/validatedependenciesversions.js]: Renamed :)

**Review by psmyrek — APPROVED** [review]
LGTM 🎉

**Pull request merged**